crates/astria-sequencer-relayer/src/sequencer_relayer.rs

```diff
@@ -145,6 +143,58 @@ impl SequencerRelayer {
     }
 }
 
+/// A controller for instructing the [`SequencerRelayer`] to shut down.
+///
+/// When constructed, it will provide a receiver to be passed into the `SequencerRelayer`'s
+/// constructor.  The `SequencerRelayer` should begin to shut down as soon as the first value is
+/// received from the receiver.
+///
+/// The controller will send the value on receiving a `SIGTERM` signal or on being dropped.
+pub struct ShutdownController {
+    sender: mpsc::Sender<()>,
+}
+
+impl ShutdownController {
+    /// Creates a new `ShutdownController`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if a signal listener could not be constructed (usually if this is not run on Unix).
+    #[must_use]
+    pub fn new() -> (Self, mpsc::Receiver<()>) {
+        let (sender, receiver) = mpsc::channel(1);
+        let cloned_sender = sender.clone();
+        let mut sigterm = signal(SignalKind::terminate())
+            .expect("setting a SIGTERM listener should always work on Unix");
+        tokio::spawn(async move {
```

Comment:
Leaving task `JoinHandle` dangling always make me a bit uneasy. This one is simple enough, but my preference is to alway `.join` or .await` them.